lib/galaxy/util/checkers.py

```diff
@@ -50,21 +53,28 @@ def check_binary(name, file_path=True):
     # Handles files if file_path is True or text if file_path is False
     if file_path:
         temp = open(name, "rb")
-        size = os.stat(name).st_size
     else:
         temp = BytesIO(name)
-        size = len(name)
-    read_start = int(size / 2)
+
     read_length = 1024
     try:
-        if util.is_binary(temp.read(read_length)):
+        buff = temp.read(read_length)
+        if len(trie.sniff(buff, False)) > 0:
+            return True
+        if util.is_binary(buff):
             return True
         # Some binary files have text only within the first 1024
         # Read 1024 from the middle of the file if this is not
         # a gzip or zip compressed file (bzip are indexed),
         # to avoid issues with long txt headers on binary files.
         if file_path and not is_gzip(name) and not is_zip(name) and not is_bz2(name):
             # file_path=False doesn't seem to be used in the codebase
+            size = os.stat(name).st_size
+            read_start = int(size / 2)
+            temp.seek(read_start)
+            return util.is_binary(temp.read(read_length))
+        if not file_path:
+            read_start = len(name) / 2
             temp.seek(read_start)
             return util.is_binary(temp.read(read_length))
```

Comment:
Do we still have a use case for this now that FCS files are handled by magic numbers?

Reply:
That is a good question.  The problem was exposed by FCS files, but are there other binary formats, without magic numbers, that might exhibit the same problem?  I will run some tests and see which of the binary test files are identified by each part of the code.

Reply:
There are no files in the test data (`lib/galaxy/datatypes/test`) that are matched by seeking to the middle of the file; all files binary files are identified (if they are) by either sniffing the magic number or looking for NULL chars in the first 1K characters.  So, it is probably safe to remove this.

Reply:
Given the performance issue with the seek, I'd suggest to go ahead with removing it. If we need to restore it at a certain point in the future, it'll still be in the git history.